### PR TITLE
VZ-4619 more VPO logging cleanup, mostly Keycloak

### PR DESCRIPTION
**Pull request by pfmackin**
More VPO logging cleanup.
Keycloak logging cleanup
Providing component info waiting for deployments, statefulsets, ingresses

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/keycloak/keycloak.go [hunk 1]
```diff
@@ -621,7 +629,7 @@ func configureKeycloakRealms(ctx spi.ComponentContext) error {
 		return err
 	}
 
-	ctx.Log().Debugf("configureKeycloakRealm: successfully configured realm %s", vzSysRealm)
+	ctx.Log().Oncef("Component Keycloak successfully configured realm %s", vzSysRealm)
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:632]: What are the semantics here?
What happens if a VZ in uninstalled and reinstalled?  Will this message ever be seen again?
Will this message be stored in a map forever?

Comment (pfmackin) [reply platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:632]: Once is in the context of a reconcile session for a resource change.  This means if you change resource foo, and the controller reconciler is called 100 times, this message will only be displayed once.  When the resource foo is finally reconciled (return ctrl.Result{}, nil to controller runtime), then we delete the logging context.  So later, if you changed the resource foo an hour later, and this code executed, then the message will be displayed once, for the new reconcile session.

Comment (kminder) [reply platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:632]: Perfect, thanks.

File: platform-operator/controllers/verrazzano/component/verrazzano/verrazzano_component.go [hunk 2]
```diff
@@ -73,7 +74,8 @@ func (c verrazzanoComponent) IsReady(ctx spi.ComponentContext) bool {
 			{Name: "verrazzano-monitoring-operator", Namespace: globalconst.VerrazzanoSystemNamespace},
 		}...)
 	}
-	if !status.DeploymentsReady(ctx.Log(), ctx.Client(), deployments, 1) {
+	prefix := fmt.Sprintf("Component %s", ComponentName)
```
Comment (mgianatagh) [inline platform-operator/controllers/verrazzano/component/verrazzano/verrazzano_component.go:77]: I have seen this line repeated many times.  Maybe a function would be better for returning the prefix.

**Review by kminder — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/externaldns/external_dns.go [hunk 3]
```diff
@@ -97,7 +98,8 @@ func isReady(compContext spi.ComponentContext) bool {
 	deployments := []types.NamespacedName{
 		{Name: externalDNSDeploymentName, Namespace: externalDNSNamespace},
 	}
-	return status.DeploymentsReady(compContext.Log(), compContext.Client(), deployments, 1)
+	prefix := fmt.Sprintf("Component %s", ComponentName)
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/externaldns/external_dns.go:101]: Are we losing the "component" log fields, so we have to do these kinds of prefixes?

**Pull request merged**